delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/RealmsSpec.scala

```diff
@@ -0,0 +1,151 @@
+        }
+      }
+
+      "reject with RealmAlreadyExists" in {
+        evaluate(wkResolution)(current, CreateRealm(label, name, wellKnownUri, None, subject))
+          .rejectedWith[RealmAlreadyExists]
+      }
+
+      "reject with RealmNotFound" in {
+        val list = List(
+          Initial -> UpdateRealm(label, 1L, name, wellKnownUri, None, subject),
+          Initial -> DeprecateRealm(label, 1L, subject)
+        )
+        forAll(list) {
+          case (state, cmd) => evaluate(wkResolution)(state, cmd).rejectedWith[RealmNotFound]
+        }
+      }
+
+      "reject with RealmAlreadyDeprecated" in {
+        evaluate(wkResolution)(current.copy(deprecated = true), DeprecateRealm(label, 1L, subject))
+          .rejectedWith[RealmAlreadyDeprecated]
+      }
+
+      "reject with wellKnown resolution error UnsuccessfulOpenIdConfigResponse" in {
+        val wellKnownWrongUri: Uri = "https://example.com/auth/realms/myrealmwrong"
+        val list                   = List(
+          Initial -> CreateRealm(label, name, wellKnownWrongUri, None, subject),
+          current -> UpdateRealm(label, 1L, name, wellKnownWrongUri, None, subject)
+        )
+        forAll(list) {
+          case (state, cmd) => evaluate(wkResolution)(state, cmd).rejectedWith[UnsuccessfulOpenIdConfigResponse]
+        }
+      }
+
+    }
+
+    "producing next state" should {
+
+      "create a new RealmCreated state" in {
+        // format: off
+        next(Initial, RealmCreated(label, 1L, name, wellKnownUri, issuer, keys, gt, None, authUri, tokenUri, uiUri, None, Some(endUri), epoch, subject)) shouldEqual
+          current.copy(createdBy = subject, updatedBy = subject)
+
+        next(current, RealmCreated(label, 1L, name, wellKnownUri, issuer, keys, gt, None, authUri, tokenUri, uiUri, None, Some(endUri), epoch, subject)) shouldEqual
+          current
+        // format: on
+      }
+
+      "create a new RealmUpdated state" in {
+        // format: off
+        next(Initial, RealmUpdated(label, 2L, name, wellKnownUri, issuer, keys, gt, None, authUri, tokenUri, uiUri, None, Some(endUri), epoch, subject)) shouldEqual
+          Initial
+
+        next(current, RealmUpdated(label, 2L, name, wellKnown2Uri, issuer, keys, gt, None, authUri, token2Uri, ui2Uri, None, Some(endUri), epoch, subject)) shouldEqual
+          Current(label, 2L, deprecated = false, name, wellKnown2Uri, issuer, keys, gt, None, authUri, token2Uri, ui2Uri, None, Some(endUri), epoch, Anonymous, epoch, subject)
+        // format: on
+      }
+
+      "create new RealmDeprecated state" in {
+        next(Initial, RealmDeprecated(label, 2L, epoch, subject)) shouldEqual Initial
+
+        next(current, RealmDeprecated(label, 2L, epoch, subject)) shouldEqual
+          current.copy(rev = 2L, deprecated = true, updatedBy = subject)
+      }
+    }
+  }
+}
+
+class MockedWellKnow(expected: Map[Uri, WellKnown]) extends WellKnownResolution {
```

Comment:
This could be used somewhere else no ? Maybe it could go in some testkit
I am also not fond of the naming including Mock, it lets think that some mocking library is under the hood. We wanted to use the term Dummy no ? It could be Stub too.

Reply:
For now I keep it here, I'll move it somewhere else when we have the need.

I agree with the change of naming though 👍